app/api/entries/[id]/route.js

```diff
@@ -0,0 +1,40 @@
+import { dbConnect } from '@/utils/dbConnects';
+import Entry from '../models/Entry';
+import { getUserFromCookie } from '@/utils/auth';
+export async function DELETE(req) {
+  await dbConnect();
+  const user = getUserFromCookie();
+  if (!user) {
+    return new Response(JSON.stringify({ error: 'Unauthorized' }), {
+      status: 401,
+    });
+  }
+  const url = new URL(req.url);
+  const entryId = url.pathname.split('/').pop();
+
+  if (!entryId) {
+    return new Response(JSON.stringify({ error: 'Entry ID is required' }), {
+      status: 400,
+    });
+  }
+  try {
+    const entry = await Entry.findById(entryId);
+    if (!entry) {
+      return new Response(JSON.stringify({ error: 'Entry not found' }), {
+        status: 404,
+      });
+    }
+    if (entry.participant.toString() != user.id) {
+      return new Response(JSON.stringify({ error: 'Forbidden' }), {
+        status: 403,
+      });
+    }
+    await Entry.findByIdAndDelete(entryId);
```

Comment:
If our goal is to find by id and delete, then maybe we don't get the entityID to begin with.
findByIdAndDelete` will return the deleted rows if the ID was found